src/collector.rs

```diff
@@ -26,29 +41,88 @@ use {
     tokio::sync::mpsc,
 };
 
+use crate::mock::MockCloudWatchClient;
+use crate::GzipSetting;
 use crate::{error::Error, BoxFuture};
 
-pub trait CloudWatch {
+pub(crate) trait CloudWatch {
     fn put_metric_data(
         &self,
         namespace: String,
         data: Vec<MetricDatum>,
     ) -> BoxFuture<'_, Result<(), SdkError<PutMetricDataError>>>;
 }
 
-impl CloudWatch for Client {
+pub(crate) struct CloudwatchClient {
+    client: Client,
+    gzip_setting: GzipSetting,
+}
+
+#[cfg(feature = "gzip")]
+#[derive(Debug)]
+struct GzipInterceptor {
+    compression: flate2::Compression,
+}
+#[cfg(feature = "gzip")]
+impl Intercept for GzipInterceptor {
```

Comment:
do we really need the Interceptor dance, or would it be enough using `customize().mutate_request(...)` ?

Reply:
i know, but then you dont need to implement this trait or use it directly (the design of the sdk is awful, the fewer explicit things from there we use the better...). mostly wonder if it results in slightly less aws boilerplate to use mutate_request vs this approach?